Modules/Module_Processing/Module_Population/xASL_wrp_GetROIstatistics.m

```diff
@@ -87,10 +92,29 @@ function xASL_wrp_GetROIstatistics(x)
 
 %% ------------------------------------------------------------------------------------------------------------
 %% 1) Load the atlas
-x = xASL_stat_AtlasForStats(x); % check all atlases that are requested (see InputAtlasPath above)
-% check them, check their ROI names, and make them ready for use below
-
+if ~x.S.bSubjectSpecificAtlas
+	x = xASL_stat_AtlasForStats(x); % check all atlases that are requested (see InputAtlasPath above)
+	% check them, check their ROI names, and make them ready for use below
 
+	% Skip atlas loading here for subject-specific atlases
+else
+	% In case of a subject specific atlas, load the subject-specific atlas for any random existing subject to be able to create the ROI names
```

Comment:
In the case of subject-specific ROIs/lesions

and remove the "random existing subjects" :)


More importantly:
It seems to me that this is too complicated.

xASL_stat_AtlasForStats does:
1. Loading atlas names, we don't have this for the lesions/ROIs, this becomes much to convolved, and these are user-specified anyway, so the user should know what ROI 1 2 3 4 etc stand for. So we don't need this for subject-wise
2. Deal with memory-mapping: we don't use memory-mapping anymore
3. Resample atlas to 1.5 isotropic, we don't need this because we have transformed the lesions & ROIs to standard space ourselves
4. Convert atlas to 4D binary image, we don't need this, each ROI or lesion is done separately (see iAtlas loop in xASL_module_Population)
5. Convert/compress masks into columns -> we currently need this, but you are removing this necessity in the other issue. So if this is the only requirement for running xASL_stat_AtlasForStats subject-wise, I would do this other issue first so we don't have to run xASL_stat_AtlasForStats.
6. Print atlas-overview image: we don't need this, because we already print a QC overview subject-wise elsewhere


CONCLUSION: 
1. I would fix the other issue first, in which you optimize the population-module NIfTI loading, and remove the conversion to a column. Then we need to remove the conversion to column as well throughout the whole xASL_module_Population in that issue, including the atlases/ROIs (because the ROIs need to be in the same format as the data anyway).
2. Then we don't need xASL_stat_AtlasForStats for subject-wise ROIs, which makes it a lot easier

Reply:
will check later....

Reply:
> In the case of subject-specific ROIs/lesions
> and remove the "random existing subjects" :)
Agree. Done.

> More importantly: It seems to me that this is too complicated.
> 
> xASL_stat_AtlasForStats does:
> 
> 1. Loading atlas names, we don't have this for the lesions/ROIs, this becomes much to convolved, and these are user-specified anyway, so the user should know what ROI 1 2 3 4 etc stand for. So we don't need this for subject-wise
> 2. Deal with memory-mapping: we don't use memory-mapping anymore
> 3. Resample atlas to 1.5 isotropic, we don't need this because we have transformed the lesions & ROIs to standard space ourselves
> 4. Convert atlas to 4D binary image, we don't need this, each ROI or lesion is done separately (see iAtlas loop in xASL_module_Population)
> 5. Convert/compress masks into columns -> we currently need this, but you are removing this necessity in the other issue. So if this is the only requirement for running xASL_stat_AtlasForStats subject-wise, I would do this other issue first so we don't have to run xASL_stat_AtlasForStats.
> 6. Print atlas-overview image: we don't need this, because we already print a QC overview subject-wise elsewhere
> 
> CONCLUSION:
> 
> 1. I would fix the other issue first, in which you optimize the population-module NIfTI loading, and remove the conversion to a column. Then we need to remove the conversion to column as well throughout the whole xASL_module_Population in that issue, including the atlases/ROIs (because the ROIs need to be in the same format as the data anyway).
> 2. Then we don't need xASL_stat_AtlasForStats for subject-wise ROIs, which makes it a lot easier

Regarding 1-6 and 1-2. I fully agree. Except for the order. We have this issue fixed already and we need it for several tumor studies ASAP. So I would finalize, test, and merge this. And then I'll immediately start working on removing the columns and all you mention above.